src/engine/world/actor/metadata.ts

```diff
@@ -0,0 +1,56 @@
+import { ConstructedRegion } from '../map';
+import { Position } from '../position';
+import { Actor } from './actor';
+
+/**
+ * The definition of the metadata available on an {@link Actor}.
+ *
+ * You cannot guarantee that all of these properties will be present on an actor,
+ * so you should always check for their existence before using them.
+ *
+ * @author jameskmonger
+ */
+export type ActorMetadata = {
+    /**
+     * The custom constructed map region for this actor.
+     *
+     * TODO (jameskmonger) Should this live on Actor rather than on {@link Player}? I don't think NPCs can have a custom map.
+     */
+    customMap: ConstructedRegion;
```

Comment:
What do you mean by customMap? Is this an instanced map that you're refering to?

Reply:
Pretty sure it should live there yes, as some npc's are instanced inside custom regions (PoH dungeon, butler in build and normal mode within the PoH). Some minigames will also use this I bet

Reply:
Stealing Creation minigame does, not sure of any others